pkg/service/system/secrets.go

```diff
@@ -69,15 +69,72 @@ func getSystemCredential() (*sts.Credentials, error) {
 	return result.Credentials, nil
 }
 
+func getSessionUsingAwsAccessKey(region string) (*session.Session, error) {
+
+	conf := config.Get()
+
+	cred := credentials.NewStaticCredentials(conf.AWSAccessID, conf.AWSSecretKey, conf.AWSToken)
+
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(region),
+		Credentials: cred,
+	})
+
+	return sess, err
+}
+
+func getSessionUsingAwsProfile(profile string) (*session.Session, error) {
+
+	sess, err := session.NewSessionWithOptions(
+		session.Options{
+			Profile: profile,
+		},
+	)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to find the credential config in system")
+	}
+
+	//NewSessionWithOptions doesn't return error if we pass the invalid profile, checking here
+	// if we got the credentials correctly
+	_, err = sess.Config.Credentials.Get()
+
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to find the credential config in system")
+	}
+
+	return sess, nil
+
+}
+
+func getLocalEnvSession(region string) (*session.Session, error) {
```

Comment:
Can we use the same funcs from `aws/session.go`? Let's avoid duplication of code.

Reply:
the duplicate is intentional, aws/session maintains the user session. Want to isolate the system session from it and not and keep the domain separate.